### PR TITLE
Move openzaak.utils.tests to openzaak.tests.utils package

### PR DESCRIPTION
**Pull request by sergei-maertens**
Periodic code maintenance

Test utilities were smeared out across `openzaak/tests/utils.py` and `openzaak/utils/tests.py`. This is now rectified and properly structured in a `openzaak/tests/utils/` package.

On top of that, the `openzaak.notifications.tests` and `openzaak.selectielijst.tests` utilities have been refactored to use `zgw_consumers.test` schema mockers and the schema's that were duplicated have been de-duplicated.

### CODE REVIEW

**Comment by sergei-maertens** [comment]
@stevenbal I hope this doesn't introduce too many merge conflicts for #1138 :grimacing:

**Comment by stevenbal** [comment]
> I hope this doesn't introduce too many merge conflicts for https://github.com/open-zaak/open-zaak/pull/1138 grimacing

We're probably not merging that until auth is properly implemented, right?

File: src/openzaak/components/besluiten/tests/test_besluit_with_external_zaak.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def test_create_with_external_zaak(self):
         url = get_operation_url("besluit_create")
 
         with requests_mock.Mocker() as m:
-            mock_service_oas_get(m, APITypes.zrc, self.base)
+            mock_service_oas_get(m, self.base, APITypes.zrc)
```
[the remark this answers is not in the document]
Comment (sergei-maertens) [reply src/openzaak/components/besluiten/tests/test_besluit_with_external_zaak.py:77]: Yep!

**Comment by sergei-maertens** [comment]
> > I hope this doesn't introduce too many merge conflicts for #1138 grimacing
> 
> We're probably not merging that until auth is properly implemented and the PR for the inclusions library is merged, right?

I think we're primarily waiting on the verdict of VNG whether it's going to be inclusions or HAL. If it's inclusions, I favour merging that PR and then adding auth to it in a separate PR. By putting it all behind the x-experimental flag we can do this without worrying about client breakage.

**Pull request merged**